WordPress/src/main/java/org/wordpress/android/ui/mysite/QuickStartRepository.kt

```diff
@@ -0,0 +1,79 @@
+package org.wordpress.android.ui.mysite
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import org.wordpress.android.fluxc.store.QuickStartStore
+import org.wordpress.android.fluxc.store.QuickStartStore.QuickStartTask
+import org.wordpress.android.fluxc.store.QuickStartStore.QuickStartTask.CREATE_SITE
+import org.wordpress.android.fluxc.store.QuickStartStore.QuickStartTaskType
+import org.wordpress.android.fluxc.store.QuickStartStore.QuickStartTaskType.CUSTOMIZE
+import org.wordpress.android.fluxc.store.QuickStartStore.QuickStartTaskType.GROW
+import org.wordpress.android.ui.quickstart.QuickStartTaskDetails
+import org.wordpress.android.util.QuickStartUtilsWrapper
+import org.wordpress.android.util.merge
+import javax.inject.Inject
+
+class QuickStartRepository
+@Inject constructor(
+    private val quickStartStore: QuickStartStore,
+    private val quickStartUtils: QuickStartUtilsWrapper,
+    private val selectedSiteRepository: SelectedSiteRepository
+) {
+    private val detailsMap: Map<QuickStartTask, QuickStartTaskDetails> = QuickStartTaskDetails.values()
+            .associateBy { it.task }
+    private val customizeTasks = MutableLiveData<QuickStartCategory>()
+    private val growTasks = MutableLiveData<QuickStartCategory>()
+    val quickStartModel: LiveData<List<QuickStartCategory>> = merge(
+            customizeTasks,
+            growTasks
+    ) { customizeCategory, growCategory ->
+        listOfNotNull(customizeCategory, growCategory)
+    }
+
+    fun startQuickStart() {
+        selectedSiteRepository.getSelectedSite()?.let { site ->
+            quickStartStore.setDoneTask(site.id.toLong(), CREATE_SITE, true)
+            refreshCustomizeTasks()
+            refreshGrowTasks()
+        }
+    }
```

Comment:
One question: how are we planning to handle site selection changes? I'm asking because I'm wondering if we should adopt a more reactive approach here later on (in another PR) by listening to `SelectedSiteRepository::selectedSiteChange` instead? Let me know your thoughts on this.

Reply:
this is a good point 👍 , I'll change the code to be more reactive